src/org/openstreetmap/josm/gui/tagging/presets/TaggingPresetSelector.java

```diff
@@ -292,11 +305,11 @@ public List<PresetClassification> getMatchingPresets(String searchText, boolean
                 nameWords = searchText.split("\\s", -1);
             }
 
-            return getMatchingPresets(groupWords, nameWords, onlyApplicable, inTags, presetTypes, selectedPrimitives);
+            return getMatchingPresets(groupWords, nameWords, onlyApplicable, inTags, isDeprecated, presetTypes, selectedPrimitives);
         }
 
-        public List<PresetClassification> getMatchingPresets(String[] groupWords, String[] nameWords, boolean onlyApplicable,
-                boolean inTags, Set<TaggingPresetType> presetTypes, final Collection<? extends OsmPrimitive> selectedPrimitives) {
+        public List<PresetClassification> getMatchingPresets(String[] groupWords, String[] nameWords, boolean onlyApplicable, boolean inTags,
+                 boolean isDeprecated, Set<TaggingPresetType> presetTypes, final Collection<? extends OsmPrimitive> selectedPrimitives) {
```

Comment:
I should probably mention (at this point) why I think we should start moving these booleans to an enum: easier readability. There are some other advantages, but `function(bool, bool, bool, bool, bool, bool)` start to get _really_ hard to wrap your mind around. Some IDEs will give hints as to what each bool is _supposed to be_, but not all IDEs will.

As a specific example, GitHub doesn't show what a boolean is supposed to be for, so the reviewer/editor has to know the function definition, or go find it.